components/plan-build/bin/hab-plan-build.sh

```diff
@@ -2059,9 +2059,9 @@ do_default_strip() {
   find "$pkg_prefix" -type f -perm -u+w -print0 2> /dev/null \
     | while read -rd '' f; do
       case "$(file -bi "$f")" in
-        *application/x-executable*) strip --strip-all "$f";;
-        *application/x-pie-executable*) strip --strip-unneeded "$f";;
-        *application/x-sharedlib*) strip --strip-unneeded "$f";;
+        *application/x-executable*) strip --strip-debug "$f";;
+        *application/x-pie-executable*) strip --strip-debug "$f";;
+        *application/x-sharedlib*) strip --strip-debug "$f";;
```

Comment:
Why the switch from `--strip-unneeded` to `--strip-debug`?

Reply:
`strip-unneeded` sounds like it would strip a lot more than `strip-debug`. I'm thinking a lot of symbols would turn into offsets which would be a lot less helpful

Reply:
Would we want `--only-keep-debug` instead? `--strip-debug` looks like it strips debug symbols.

Reply:
I think `--only-keep-debug` will give you something that is not executable. `--strip-debug` is a nice compromise because it takes a little more space to give you some human readable symbols, but you wont get all the goodness like lineinfo. Now, if we could come up with a way to make publish multiple packages from 1 build, then we could do something crazy like publish completely stripped binaries, and have the debug info live in another package that we could grab when we needed it.

Reply:
It's worth considering that this change is going to affect all new builds for all packages, so it's worth considering that. We can also just do this for our packages for now (hab, hab-sup, hab-launch, etc)

Reply:
The point of `--only-keep-debug` is to create seperate debug files that gdb can load in parallel.  So definitely not what we wold want here.